src/main/java/org/cloudfoundry/promregator/endpoint/MetricsEndpoint.java

```diff
@@ -63,6 +67,10 @@ protected boolean isLabelEnrichmentSuppressable() {
 	 */
 	@Override
 	protected List<MetricsFetcher> createMetricsFetchers(List<Instance> instanceList) {
+		if(maxAllowedForSingleEndpoint != -1 && instanceList.size() > maxAllowedForSingleEndpoint) {
+			throw new RuntimeException("promregator.single.endpoint.max is configured to only allow "+maxAllowedForSingleEndpoint+" instances but there were "+ instanceList.size());
```

Comment:
This will cause a SonarQube finding. Creating a dedicated exception for this is recommended then (may be private and local to the class here).